include/instantiate.h

```diff
@@ -410,6 +410,78 @@ namespace quda
     }
   }
 
+  /**
+     @brief This instantiate function is used to instantiate combinations of reconstruct
+     and phase for pure-gauge routines using staggered phases
+     @param[in] U Gauge field
+     @param[in,out] args Additional arguments for kernels
+  */
+  template <template <typename, int, QudaReconstructType, QudaStaggeredPhase> class Apply, typename store_t, int nColor, typename G, typename... Args>
+  constexpr void instantiateGaugeHisq(G &U, Args &&...args)
```

Comment:
Given the description, perhaps this should be named `instantiateGaugeStaggered`.  There's nothing specifically HISQ here is there?

Reply:
Ultimately this will be for recon-18/13/12 (once I get that working), which is why I made it separate. I can add a comment to this sort, or a separate error path with a `recon-12`-specific error.

Reply:
https://github.com/lattice/quda/pull/1367/commits/49435f4ff06d1e6ddd5d04699d2922e08a21e016 --- let me know if this is good

Reply:
My point was here that the function name `instantiateGaugeHisq` isn't the best name, and something like `instantiateGaugeStaggered` would be more descriptive, since it instantiates templates for reconstruct and staggered phase (as well as color and precision).

Reply:
Ah, okay, understood

Reply:
https://github.com/lattice/quda/pull/1367/commits/26d81fcf5af38cab8913e5ca750be024c98eed84